source/Outfit.cpp

```diff
@@ -88,6 +87,31 @@ namespace {
 				thisMap.erase(it.first);
 		}
 	}
+	
+	// Summation that avoids loss of trailing digits as much as possible.
+	double Sum(const map<double, int> &partialTerms)
+	{
+		vector<double> partialSums;
+		partialSums.reserve(partialTerms.size());
+		for(const auto &it : partialTerms)
+			partialSums.push_back(it.first * it.second);
+		sort(partialSums.begin(), partialSums.end(), [&](double a, double b){ return fabs(a) < fabs(b); });
```

Comment:
If we want to avoid loss of trailing digits as much as possible, then I would expect the smallest numbers to be accumulated first before the large numbers. Because the loss mostly happens when small numbers are added to large numbers.